term.go

```diff
@@ -285,13 +285,10 @@ func (t *Terminal) run() {
 			fyne.LogError("pty read error", err)
 		}
 
-		lenLeftOver := len(leftOver)
-		if lenLeftOver > 0 {
-			buf = append(leftOver, buf[:num]...)
-			num += lenLeftOver
-		}
-		leftOver = t.handleOutput(buf[:num])
-		if num < bufLen+lenLeftOver {
+		leftOver = append(leftOver, buf[:num]...)
```

Comment:
For what I found with using append.
The append function tries to add the elements to the existing slice if there's enough capacity. If not, it creates a new slice with double the capacity (or more if needed), copies the existing elements, and adds the new elements. We avoid this by setting leftOver to bufLen + 3 for the largest UTF-8 partial possible.

Reply:
This does sound like a sensible strategy - the summary of append is correct in my understanding too.